test/unitary/xtypes/collection_types.cpp

```diff
@@ -96,6 +96,41 @@ TEST (CollectionTypes, multi_array)
 
 }
 
+TEST (CollectionTypes, multi_array_constructor)
+{
+
+    ArrayType array_array_array(primitive_type<uint32_t>(), {3, 4, 5});
+    const ArrayType& inner = static_cast<const ArrayType&>(array_array_array.content_type());
+    const ArrayType& inner_inner = static_cast<const ArrayType&>(inner.content_type());
+
+    EXPECT_EQ(array_array_array.dimension(), 3);
+    EXPECT_EQ(inner.dimension(), 4);
+    EXPECT_EQ(inner_inner.dimension(), 5);
+
+    EXPECT_EQ(array_array_array.memory_size(), 3 * 4 * 5 * sizeof(uint32_t));
+
+    DynamicData data(array_array_array);
+
+    EXPECT_EQ(data.bounds(), 3);
+
+    for (size_t i = 0; i < 3; ++i)
+    {
+        EXPECT_EQ(data[i].bounds(), 4);
+        for (size_t j = 0; j < 4; ++j)
+        {
+            EXPECT_EQ(data[i][j].bounds(), 5);
+            for (size_t k = 0; k < 5; ++k)
+            {
+                data[i][j][k] = static_cast<uint32_t>(i + j + k);
+                uint32_t temp = data[i][j][k];
+                EXPECT_EQ(data[i][j][k].value<uint32_t>(), static_cast<uint32_t>(i + j + k));
```

Comment:
I think it is better to check the values in another loop. Maybe we detect that setting a value could  reverberate in previous settings.

Reply:
TOTALLY agree.